agent/config/config.go

```diff
@@ -225,26 +227,28 @@ func EnvironmentConfig() Config {
 	}
 }
 
+var ec2MetadataClient = ec2.DefaultClient
+
 func EC2MetadataConfig() Config {
-	iid, err := ec2.GetInstanceIdentityDocument()
+	iid, err := ec2MetadataClient.InstanceIdentityDocument()
 	if err != nil {
 		log.Crit("Unable to communicate with EC2 Metadata service to infer region and endpoint: " + err.Error())
```

Comment:
Change this message to say infer region.  Endpoint is not inferred here anymore.